Requires individual components to pluralize

AlexSCorey opened the pull request:
##### SUMMARY
This addresses https://github.com/ansible/awx/issues/4705.  We have had issues in the past in awx-pf with pluralization (only 1 or 2).  This PR requires that the components that use `PaginatedDataList` and `DataListToolBar` provide the pluralized `itemName` and removes `pluralize()` from `string utils`.  I feel this is an improvement because the `pluralize()` is not very effective.  After all, how to pluralize a word can be quite complicated (https://www.englishgrammar.org/rules-formation-plurals/).  Also it was used inside `PaginatedDataList` and `DataListToolBar` and neither component should care whether the `itemName` needs to be pluralized.  This method requires the component that uses `PaginatedDataList` and `DataListToolBar` to properly pass `itemName` along in the appropriate form (singular/plural).

If we are on board with this I will add some tests before merging this.  

Another alternative would be to import a pluralization library.  Not so sure we need that, though.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
7.0.0
```
##### ADDITIONAL INFORMATION
I believe I have addressed all of the places that used `pluralize()` but I might have missed a place here or there that isn't built out yet.

AlexSCorey committed:
requires individual components to pluralize

keithjgrant commented on awx/ui_next/src/components/PaginatedDataList/PaginatedDataList.jsx:
```diff
@@ -82,10 +81,8 @@ class PaginatedDataList extends React.Component {
-      itemNamePlural || pluralize(itemName)
-    );
+    const itemDisplayName = ucFirst(itemName);
+    const itemDisplayNamePlural = ucFirst(itemName);
```
This is the same as the singular itemDisplayName. We need to either pass in both singular & plural words as props, or put a getter function on props, something like:

```
<PaginatedDataList
  ...
  getItemName={isPlural => isPlural ? 'Jobs' : 'Job'}
/>

... we can probably also get rid of the `ucFirst` here, since we're only passing in & using capitalized versions, too.

AlexSCorey replied:
@keithjgrant perhaps I don't understand what you are saying but `itemName` is a word(s) in singular or plural form that is passed in from the component using PDL. That component determines the plurality of `itemName` so I don't quite understand why we need to pass both singular and plural words.  I do think, though, that I can get ride of both `itemDisplayName` and `itemDisplayNamePlural` as well as `ucFirst()`

keithjgrant replied:
my point here is you have two variables with the same value. so get rid of the second one and just use the first one throughout... In fact, you can get rid of both `itemDisplayName` and `itemDisplayNamePlural`, and just use `itemName` throughout, since it's already capitalized by the parent components

keithjgrant commented on awx/ui_next/src/screens/Job/JobList/JobList.jsx:
```diff
@@ -160,7 +160,7 @@ class JobList extends Component {
             items={jobs}
             itemCount={itemCount}
-            itemName={itemName}
+            itemName={itemCount === 1 ? i18n._(t`Job`): i18n._(t`Jobs`)}
```
We can't translate these here, because it needs to go inside a full sentence message inside PDL; then the full sentence gets translated as a whole by PDL. Same with other List components below

AlexSCorey replied:
Good catch here!

keithjgrant commented on awx/ui_next/src/components/PaginatedDataList/PaginatedDataList.jsx:
```diff
@@ -82,10 +81,8 @@ class PaginatedDataList extends React.Component {
+    const itemDisplayName = ucFirst(itemName);
+    const itemDisplayNamePlural = ucFirst(itemName);
 
     const dataListLabel = i18n._(t`${itemDisplayName} List`);
```
Note that these messages should NOT change depending on the number of items in the list. They should always be as coded here, e.g. "Job List" (singular), "Please add Jobs to populate this list" (plural), "No Jobs found" (plural).

HOWEVER -- we might be able to convert all to plural and still have it makes sense (Switch to "Jobs List"). Then we would only need the plural version of the word and could cut out a lot of this complexity. 🤔

AlexSCorey replied:
I'll investigate whether we can make all of these phrases plural to see if we can cut down on the complexity.

AlexSCorey replied:
So the way to code currently works is that if itemName is `!1` then itemName is plural.  So, if the `Job List` contains only 1 job then it would render as `Job List` otherwise there will be more than 1 job and it will be `Jobs List`, which makes sense to me.  However, if we always want it to be in the plural that can be done too.

keithjgrant reviewed (changes requested):
I think this is a good start at simplifying things. The way it's setup now, though, items names are getting run through translation twice. Added some comments below.

AlexSCorey committed:
Adds proper translation.  removes ucFirst()>

AlexSCorey committed:
Pluralized modal titles and empty state strings

All itemNames used in empty state messages, and delete modal titles
need to be plural and capitalized.  Because of this change we no
longer need the ucFirst() in utils/strings.js.

keithjgrant reviewed (approved):
Looks good!

kialam commented:
regate
